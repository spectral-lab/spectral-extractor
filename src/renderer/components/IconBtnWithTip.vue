<template>
  <v-tooltip bottom>
    <template #activator="{ on }">
      <v-btn
        v-on="on"
        @click="emitClickEvent"
        small
        fab
        color="primary"
        class="elevation-0"
      >
        <v-icon>{{ icon }}</v-icon>
      </v-btn>
    </template>
    <span>{{ tip }}</span>
  </v-tooltip>
</template>

<script>
import Vue from 'vue';
export default Vue.extend({
  props: {
    icon: String,
    tip: String
  },
  methods: {
    emitClickEvent () {
      this.$emit('click');
    }
  }
});
</script>
