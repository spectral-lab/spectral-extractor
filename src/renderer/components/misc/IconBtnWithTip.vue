<template>
  <v-tooltip bottom>
    <template #activator="{ on }">
      <v-btn
        v-on="on"
        @click="emitClickEvent"
        :color="color"
        small
        fab
        class="elevation-0"
      >
        <v-icon>{{ icon }}</v-icon>
      </v-btn>
    </template>
    <span>{{ tip }}</span>
  </v-tooltip>
</template>

<script>
import Vue from 'vue';
export default Vue.extend({
  props: {
    icon: {
      type: String,
      default: ''
    },
    tip: {
      type: String,
      default: ''
    },
    color: {
      type: String,
      default: 'primary'
    }
  },
  methods: {
    emitClickEvent (ev) {
      this.$emit('click', ev);
    }
  }
});
</script>
