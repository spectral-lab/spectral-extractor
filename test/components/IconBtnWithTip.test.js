import { shallowMount } from '@vue/test-utils';
import IconBtnWithTip from '../../src/renderer/components/IconBtnWithTip';
import Vue from 'vue';
import vuetify from 'vuetify';

Vue.use(vuetify);

describe('Component', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(IconBtnWithTip, {
      propsData: {
        icon: 'dashboard',
        tip: 'hello world'
      }
    });
  });
  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
  });
  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
