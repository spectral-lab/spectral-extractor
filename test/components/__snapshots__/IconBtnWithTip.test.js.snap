// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders correctly 1`] = `
<v-tooltip-stub
  bottom="true"
  closedelay="200"
  contentclass=""
  debounce="0"
  fixed="true"
  maxwidth="auto"
  nudgebottom="0"
  nudgeleft="0"
  nudgeright="0"
  nudgetop="0"
  nudgewidth="0"
  opendelay="200"
  tag="span"
>
   
  <span>
    hello world
  </span>
</v-tooltip-stub>
`;
